etna/whatson/models.py

```diff
@@ -0,0 +1,633 @@
+        help_text=_("The type of venue for the event."),
+    )
+
+    venue_website = models.URLField(
+        max_length=255,
+        verbose_name=_("venue website"),
+        null=True,
+        blank=True,
+        help_text=_("The website for the venue."),
+    )
+
+    venue_address = RichTextField(
+        verbose_name=_("venue address"),
+        null=True,
+        blank=True,
+        help_text=_("The address of the venue."),
+    )
+
+    venue_space_name = models.CharField(
+        max_length=255,
+        verbose_name=_("venue space name"),
+        null=True,
+        blank=True,
+        help_text=_("The name of the venue space."),
+    )
+
+    video_conference_info = RichTextField(
+        verbose_name=_("video conference info"),
+        null=True,
+        blank=True,
+        help_text=_("Information about the video conference."),
+    )
+
+    # Booking information
+    booking_type = models.CharField(
+        max_length=20,
+        verbose_name=_("booking type"),
+        null=False,
+        blank=False,
+        default="Drop in",
+        help_text=_("The type of booking for the event."),
+    )
+
+    registration_url = models.URLField(
+        max_length=255,
+        verbose_name=_("registration url"),
+        null=True,
+        blank=True,
+        help_text=_("The URL for the event registration."),
+    )
+
+    registration_cost = models.IntegerField(
+        verbose_name=_("registration cost"),
+        null=True,
+        blank=True,
+        help_text=_("The cost of registration for the event."),
+    )
+    # The three fields above will be brought in from the API when we have it.
+
+    registration_info = RichTextField(
+        verbose_name=_("registration info"),
+        null=True,
+        blank=True,
+        help_text=_("Information about how to register for the event."),
+    )
+
+    contact_info = RichTextField(
+        verbose_name=_("contact info"),
+        null=True,
+        blank=True,
+        help_text=_("Information about who to contact for the event."),
+    )
+
+    # Promote tab
+    short_title = models.CharField(
+        max_length=50,
+        verbose_name=_("short title"),
+        help_text=_(
+            "A short title for the event. This will be used in the event listings."
+        ),
+        blank=False,
+        null=True,
+    )
+
+    # DataLayerMixin overrides
+    gtm_content_group = "What's On"
+
+    class Meta:
+        verbose_name = _("Event page")
+
+    content_panels = BasePageWithIntro.content_panels + [
+        FieldPanel("lead_image"),
+        MultiFieldPanel(
+            [
+                FieldPanel("event_type"),
+                FieldPanel("start_date", read_only=True),
+                FieldPanel("end_date", read_only=True),
+                InlinePanel(
+                    "event_sessions",
+                    heading=_("Event sessions"),
+                    help_text=_("List of event sessions, ordered from first to last."),
```

Comment:
Ordering doesn't matter to the editor, technically.... we'll always display them in date order